From: SeongJae Park <sj@kernel.org>
Date: Fri, 22 Nov 2024 12:15:16 -0800
Subject: [PATCH] mm/damon/paddr: support DAMOS filter type UNMAPPED

DAMOS filter type of UNMAPPED is defined, but no DAMON operations set is
supporting it.  Implement the support on 'paddr', the DAMON operations
set for the physical address space.

Signed-off-by: SeongJae Park <sj@kernel.org>
---
 mm/damon/paddr.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/mm/damon/paddr.c b/mm/damon/paddr.c
index 4733aaddcae8..b3cb2578a201 100644
--- a/mm/damon/paddr.c
+++ b/mm/damon/paddr.c
@@ -222,6 +222,9 @@ static bool __damos_pa_filter_out(struct damos_filter *filter,
 		if (matched)
 			damon_folio_mkold(folio);
 		break;
+	case DAMOS_FILTER_TYPE_UNMAPPED:
+		matched = !folio_mapped(folio) || !folio_raw_mapping(folio);
+		break;
 	default:
 		break;
 	}
-- 
2.39.5

